### ISSUE TITLE
Consider using syncpack as replacement for script ensuring package versions match up

### ISSUE THREAD

**Issue by goldstack** [goldstack/goldstack#80]
Syncpack supports [yarn workspaces](https://github.com/JamieMason/syncpack/issues/20) so it maybe a possible replacement for fixing project references https://github.com/goldstack/goldstack/blob/master/package.json#L20

Just need to see if it can supported nested Yarn 2/Berry workspaces.

**Comment by JamieMason** [comment]
syncpack creator here, syncpack just needs glob patterns for where your package.json files are, so the exact structure of your repo won't matter. The Yarn/Lerna etc defaults are there as fallbacks for convenience when no glob patterns have been provided.

**Comment by mxro** [comment]
A very nice, I saw https://github.com/JamieMason/syncpack#source, according to which it would be here: `workspaces/*/packages*/package.json`.

What I am particularily after is to ensure that the local packages defined in the monorepo are referenced by packages depending on them with exactly the correct version. To ensure that these packages use the local dependency rather than download a package from npm, so.

**Package Used**
 
```
{
   "name": "lib-in-monorepo",
   "version": "0.2.1",
}
```

**Package using**

```
{
   "name": "app-in-monorepo",
   "version": "1.01",
   "dependencies": {
       "lib-in-monorepo": "0.2.1",
    }
}
```

Currently this is implemented as https://github.com/goldstack/goldstack/blob/master/workspaces/templates-lib/packages/utils-package-upgrade/src/utilsPackageUpgrade.ts#L12

**Comment by JamieMason** [comment]
Got you, that feature's not available just yet but there's an open issue here if you wanted to subscribe https://github.com/JamieMason/syncpack/issues/66. Thanks a lot

**Comment by mxro** [comment]
Awesome, subscribed to the issue and will keep an eye on it 👍

**Comment by mxro** [comment]
Also see https://github.com/goldstack/goldstack/issues/11

**Comment by JamieMason** [comment]
Hi all, this was released in [syncpack@7.0.0](https://github.com/JamieMason/syncpack/releases/tag/7.0.0). If you decide to give it a go, please shout if you run into any problems.

**Comment by mxro** [comment]
Works using `syncpack@8` and merged.
